### PR TITLE
project: read local plugins from build-aux

### PR DESCRIPTION
**Pull request by cmatsuoka**
When loading snapcraft.yaml from the build-aux/snap instead of snap, we
should also read local plugins from the alternative location.

LP: #1827095

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/project/_project.py [hunk 1]
```diff
@@ -53,14 +54,28 @@ def __init__(
 
         super().__init__(target_deb_arch, debug, work_dir=work_dir)
 
+        self.local_plugins_dir = self._get_local_plugins_dir()
+
     def _get_snapcraft_assets_dir(self) -> str:
+        if not self.info:
```
Comment (sergiusens) [inline snapcraft/project/_project.py]: Can you add a comment on when this situation takes place?

File: snapcraft/internal/pluginhandler/_plugin_loader.py [hunk 2]
```diff
@@ -29,10 +30,19 @@
...
 
 def load_plugin(
-    plugin_name, part_name, project_options, properties, part_schema, definitions_schema
+    plugin_name,
+    part_name,
+    project_options,
+    properties,
+    part_schema,
+    definitions_schema,
+    local_plugins_dir=None,
```
Comment (sergiusens) [inline snapcraft/internal/pluginhandler/_plugin_loader.py]: mind adding type annotation for this new entry? Optional[str] I believe it is...

This final comma here might break flake8, let's see from the tests.

File: snapcraft/project/_project.py (hunk 1 above)
Comment (cmatsuoka) [reply snapcraft/project/_project.py]: Comment added.

File: snapcraft/internal/pluginhandler/_plugin_loader.py (hunk 2 above)
Comment (cmatsuoka) [reply snapcraft/internal/pluginhandler/_plugin_loader.py]: Added type annotation.

**Pull request merged**